Massdrop/fluxible-router
# Jxliu/0.4.x/navlink onclick fix

Asana: https://app.asana.com/0/33626641634273/116679641959726

createNavLinkComponent does not handle the case where we provide an onClick via props. Instead of merging the onClick props call with the default dispatchNavAction onClick, the onClick props overrides the dispatchNavAction behavior.

This fix allows both to coexist. Otherwise whenever we add an onClick to a NavLink, there will be a full page load instead of an internal navigation.

This fix was copied from MOP:
https://github.com/Massdrop/fluxible/commit/6e42f0648cc1c735d05c476253f8c818946c2b54

#### Testing Instructions

1. Modify this line in your massdrop-react package.json: ```"fluxible-router": "git@github.com:Massdrop/fluxible-router.git#53b4d3b515c4002f37531fa6070a0b448c092792",```
2. ```npm install```
3. Go to a drop page and click the Join or Commit button. You should be taken to the checkout page without a full page load. A couple indications that there was not a full page load: no browser loading bar, console log not cleared (disable preserve log in Chrome), generally feels fast.

After merging, I'll handle the publish and pinning react to the new fluxible-router version.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/massdrop/fluxible-router/5)
